### DISCUSSION TITLE
How to sign up fo an API in Apiman Portal

### DISCUSSION THREAD

**Discussion by ronimhd** [apiman/apiman#2572] [Q&A]
Hi,
I am trying Apiman 3.1.2.
I have a doubt regarding who can sign up to an API in Apiman Developer Portal. 
I this case, I have created a user with "API-Management-Devportal-Users".

![image](https://github.com/apiman/apiman/assets/25297110/2aac8b78-bef2-4cea-b661-62a7fe726fb1)

And then, I created an API with discoverability to Developer Portal users:

![image](https://github.com/apiman/apiman/assets/25297110/75063497-c0bf-4142-9c7a-a0031598933b)

I thought, this user will be able to sign up to the API. But turns out, the user will get this error message:

![image](https://github.com/apiman/apiman/assets/25297110/c5203b69-957e-45e8-b233-9d1e912b121e)

Then I tried again with some conditions, only if the user is a member of an Organization, and has ClientApp role, the user can sign up.
So, it is by designed? but why the Sign-Up button still available for the user.  
Or anything that I need to check in my Apiman configuration?

**Comment by volkflo** [comment]
Devportal role should be enough. Otherwise these people would be able to login into the normal ui.
We create an organization for these users automatically during client creation if the user does not have one.
The error indicates that there is something wrong as the org is missing in the URL.
Probably it could not be created or something like that.

You have to provide us the browsers console log and backend logs.
Otherwise we are not able to help.

**Reply by msavy** [reply]
org segment of the URL in the error seems to be blank (double forward slash).

**Reply by volkflo** [reply]
Yep, I guess there was an error in the step before.
Hard to tell without browser logs.

**Reply by volkflo** [reply]
https://github.com/apiman/apiman/blob/c4419395ba2b5858304bd21b81fb2639af2a545b/portal/ui/src/app/components/marketplace-client-app/marketplace-client-app.component.ts#L114-L132

**Reply by ronimhd** [reply]
I am sorry for late reply. Here are the console log:

![image](https://github.com/apiman/apiman/assets/25297110/d81d97a1-354e-4748-8265-5ec2e531da38)

**Reply by volkflo** [reply]
Thanks, Please avoid using screenshots of logs in the future and instead paste it into a file.

I have still one thing in my head we might not thought about it.

Can you please provide me what roles you have configured in apiman. (apimanui/api-manager/admin/roles).
Current apiman needs to auto-grant every org `clientEdit` permissions. Otherwise the portal may not able to fetch the infos it needs.

In addition it would be very helpful if you could reproduce the thing again and generate us the HAR file like explained in the link below:
https://auth0.com/docs/troubleshoot/troubleshooting-tools/generate-and-analyze-har-files#google-chrome
